src/herbpy/action/rogue.py

```diff
@@ -0,0 +1,321 @@
+import logging, openravepy, prpy 
+from prpy.action import ActionMethod
+from prpy.planning.base import PlanningError
+from contextlib import contextmanager 
+from prpy.util import FindCatkinResource
+import numpy, cPickle, time, os.path
+
+logger = logging.getLogger('herbpy')
+
+@ActionMethod
+def Point(robot, focus, manip=None, render=False):
+    """
+    @param robot The robot performing the point
+    @param focus The 3-D coordinate in space or object 
+                 that is being pointed at
+    @param manip The manipulator to perform the point with. 
+                 This must be the right arm
+    @param render Render tsr samples during planning
+    """
+
+    #Pointing at an object
+    if type(focus) == openravepy.openravepy_int.KinBody:
+        focus_trans = focus.GetTransform()
+        goal_name = focus.GetName()
+    #Pointing at a point in space
+    elif (type(focus) == numpy.ndarray) and (focus.ndim == 1):
+        if len(focus) != 3:
+            raise prpy.exceptions.PrPyExceptions('A point in space must \
+                    contain exactly three coordinates')
+        focus_trans = numpy.eye(4)
+        focus_trans[0:3, 3] = focus
+        goal_name = None
+    else:
+        raise prpy.exceptions.PrPyException('Focus of the point is an \
+                unknown object')
+
+    if manip is None:
+        print "Setting the right arm to be the active manipulator."
+        manip = robot.right_arm
+        manip.SetActive()
```

Comment:
This changes the state of the active manipulator. You should wrap this, and any code that relies on the manipulator being active, in a `RobotStateSaver` with the appropriate flags set.